### ISSUE TITLE
[BUG] _modules/ do not override native module on Windows

### ISSUE THREAD

**Issue by amalaguti** [saltstack/salt#58471]
**Description**
I put a copy of file.py module in /srv/salt/_modules, did a litlle change on the module /srv/salt/_modules/file.py (something simple like adding a log.info line to some function)
sync'd module to minions, confirmed all minions got the custom module
Windows minion still executed the native module, while in Linux minions the response was from the custom module

Looks like do not override native modules on Windows

**Steps to Reproduce the behavior**
Copy a native module to /srv/salt/_modules
cp /usr/lib/python3.6/site-packages/salt/modules/file.py /srv/salt/_modules/

Make a change to a function, synchronize the module to Windows minion and execute.
Verify if the change has been takne by Windows minion

**Expected behavior**
Windows minion should use custom module

**Versions Report**
3001.1 py3

**Comment by lukasraska** [comment]
@amalaguti it seems `file.py` module is used only for non-Windows OSes (https://github.com/saltstack/salt/blob/master/salt/modules/file.py#L84 ), I guess you need to either update the virtual method to allow Windows, or probably better option, do the same change to `win_file.py` asa well (https://github.com/saltstack/salt/blob/master/salt/modules/win_file.py)

**Comment by amalaguti** [comment]
@lukasraska I thought the same, but if you take a look to win_file, it imports functions from file.py https://github.com/saltstack/salt/blob/master/salt/modules/win_file.py#L93
Specifically I'm interested in get_managed() function.
I'm doing a little change to redact the message when there's an exception
https://github.com/saltstack/salt/blob/master/salt/modules/file.py#L4595 
I have changed file.py directly on Windows file system and it took the change. So I know for sure file.py is being read and used by Windows.

Just in case, I have put a copy of win_file.py in _modules/, sync'd the module to Windows, but still the same issue. The get_managed() function used is the one from the native file.py instead of the one in _modules/

**Comment by lukasraska** [comment]
Yeah, since it does direct import from `salt.modules.file` and doesn't go through loader, there isn't much you can do. You could just create your custom module (with different name, but `file` as virtual name) with `get_managed` copied from `file`, along with your modifications. That way it should have preference when using it as Salt execution module. Otherwise I don't see any easier option, but maybe somebody from Salt core team can think of something.

When you sync your custom module, it's available only through Salt loader (so e.g. `__salt__["config.get"](...)`), you cannot replace the original file to be available through Python imports.

**Comment by amalaguti** [comment]
I tried using virtual in my custom module to override the function but still no luck. 
```
__virtualname__ = "file"
def __virtual__():
    return __virtualname__
```

The file.function used is the one from the native file.py. I can tell immediately by renaming the function in the native module, then the minion responds wits the custom one from _modules.

**Comment by xeacott** [comment]
So are you able to write a custom module, like a `hello_world.py` or something and confirm you're able to sync that to Window's minion? Just seeing if there's a problem with syncing to a Windows minion or if this specifically has to do with the fact that `file.py` and `win_file.py` exist.

**Comment by amalaguti** [comment]
@xeacott yes Joe, I can sync modules to Windows.
